addons/l10n_it_edi_sdicoop/models/account_edi_format.py

```diff
@@ -224,28 +233,31 @@ def _l10n_it_post_invoices_step_2(self, invoices):
                     During the following 15 days, the Exchange System will try to forward the FatturaPA\
                     file to the Administration in question again.'),
                 }
-            elif state == 'notificaEsito':
+            elif state == 'notificaEsito' and response.get('file'):
                 outcome = response_tree.find('Esito').text
                 if outcome == 'EC01':
-                    to_return[invoice] = {'attachment': invoice.l10n_it_edi_attachment_id}
+                    to_return[invoice] = {'attachment': invoice.l10n_it_edi_attachment_id, 'success': True}
                 else:  # ECO2
                     to_return[invoice] = {'error': _('The invoice was refused by the addressee.'), 'blocking_level': 'error'}
             elif state == 'NotificaDecorrenzaTermini':
-                to_return[invoice] = {'error': _('Expiration of the maximum term for communication of acceptance/refusal'), 'blocking_level': 'error'}
-            proxy_acks.append(id_transaction)
+                to_return[invoice] = {'attachment': invoice.l10n_it_edi_attachment_id, 'success': True}
 
-        try:
-            proxy_user._make_request(server_url + '/api/l10n_it_edi/1/ack',
-                                     params={'transaction_ids': proxy_acks})
-        except AccountEdiProxyError as e:
-            # Will be ignored and acked again next time.
-            _logger.error('Error while acking file to SdiCoop: %s', e)
+            if response.get('file'):
+                proxy_acks.append(id_transaction)
+
+        if proxy_acks:
+            try:
+                proxy_user._make_request(server_url + '/api/l10n_it_edi/1/ack',
+                                        params={'transaction_ids': proxy_acks})
+            except AccountEdiProxyError as e:
+                # Will be ignored and acked again next time.
+                _logger.error('Error while acking file to SdiCoop: %s', e)
 
         return to_return
 
     def _post_fattura_pa(self, invoices):
         # OVERRIDE
-        if not invoices.l10n_it_edi_transaction:
+        if not invoices[0].l10n_it_edi_transaction:
```

Comment:
This is very suspicious, why is it done?

Reply:
because you can have multiple invoices in the same batch and it gives a traceback when there are multiple invoices

Reply:
Yes, but which traceback? Because this shouldnt traceback, even if there are multiple invoices and multiple edi transactions.
```python
if not invoices.l10n_it_edi_transaction:
```

Reply:
edi_result = edi_format._post_invoice_edi(documents.move_id, test_mode=test_mode)
  File "/home/odoo/source/enterprise/l10n_pe_edi/models/account_edi_format.py", line 732, in _post_invoice_edi
    return super()._post_invoice_edi(invoices, test_mode=test_mode)
  File "/home/odoo/source/odoo/addons/l10n_it_edi/models/account_edi_format.py", line 170, in _post_invoice_edi
    return self._post_fattura_pa(invoices)
  File "/home/odoo/source/odoo/addons/l10n_it_edi_sdicoop/models/account_edi_format.py", line 261, in _post_fattura_pa
    if not invoices.l10n_it_edi_transaction:
  File "/home/odoo/source/odoo/odoo/fields.py", line 965, in __get__
    record.ensure_one()
  File "/home/odoo/source/odoo/odoo/models.py", line 5000, in ensure_one
    raise ValueError("Expected singleton: %s" % self)
ValueError: Expected singleton: account.move(21, 22, 24, 25, 26, 28, 29, 30, 31, 32, 34, 35, 36, 37, 38, 39)

@william-andre

Reply:
Ok, so we can do that thanks to `_get_batch_key`; I don't know why I thought it was a relation field